concepts/Common/Integration Guides/70 Manage Azure Blob Storage with DevExtreme Components/30 FileManager.md

```diff
@@ -0,0 +1,203 @@
+                isDirectory: true,
+            };
+            directories[dirName] = directory;
+            result.push(directory);
+            }
+            if (!directory.hasSubDirectories) {
+            directory.hasSubDirectories = parts.length > 2;
+            }
+        }
+        });
+        result.sort(this.compareDataObjects);
+        return result;
+    }
+
+It outputs FileManager-compatible JSON:
+
+    [
+        {
+            "name": "folder12",
+            "isDirectory": true,
+            "hasSubDirectories": true
+        },
+        ... other records
+        {
+            "name": "testflowers_137KB.jpgd1578202-9d72-4806-9ac2-90551d598e70",
+            "isDirectory": false,
+            "dateModified": "2022-12-21T14:57:55.000Z",
+            "size": 140998
+        }
+    ]
+
+
+### Server-side Data Handling
+
+If we want to process Blob data on the server, we need to create a class that implements the interface requirements of the FileManager component:
+
+    public class AzureBlobFileProvider : IFileSystemItemLoader, IFileSystemItemEditor, IFileUploader, IFileContentProvider {
+        ...
+        public AzureBlobFileProvider(string storageAccountName, string storageAccessKey, string containerName, string tempDirPath) {
+            ...
+        }
+    ...
+    }
+
+And create function implementations for the Azure API:
+
+    public IEnumerable<FileSystemItem> GetItems(FileSystemLoadItemOptions options) {
+        var result = new List<FileSystemItem>();
+        string dirKey = GetFileItemPath(options.Directory);
+        var oneLevelItemsList = GetOneLevelHierarchyBlobs(dirKey);
+        foreach(BlobHierarchyItem hierarchyItem in oneLevelItemsList) {
+            var fileItem = GetFileSystemItem(hierarchyItem);
+            if(fileItem != null) {
+                result.Add(fileItem);
+            }
+        }
+        return result.OrderByDescending(item => item.IsDirectory)
+            .ThenBy(item => item.Name)
+            .ToList();
+    }
+
+The `FileSystem` end point will expose an instance of the `AzureBlobFileProvider` class to our client-side application:
+
+    [Route("api/file-manager-azure", Name = "FileManagerAzureProviderApi")]
+    public object FileSystem(FileSystemCommand command, string arguments) {
+        var config = new FileSystemConfiguration {
+            Request = Request,
+            FileSystemProvider = AzureFileProvider,
+            ...
+            UploadConfiguration = new UploadConfiguration {
+                MaxFileSize = 1048576
+            },
+            TempDirectory = WebHostEnvironment.ContentRootPath + "/UploadTemp"
+        };
+        var processor = new FileSystemCommandProcessor(config);
+        var result = processor.Execute(command, arguments);
+        return result.GetClientCommandResult();
+    }
+
+This approach greatly simplifies component setup. We can create a `RemoteFileSystemProvider` that uses the newly created filesystem endpoint:
```

Comment:
1. Please change "approach" to "technique".

2. Change the first person plural "We" to "You".

3. See my comment above regarding the spelling of "filesystem".